### DISCUSSION TITLE
How to use the `escaped` combinator?

### DISCUSSION THREAD

**Discussion by nicopap** [winnow-rs/winnow#242] [Q&A] [answered]
### What I parse

```
foo \| bar
```

### My parser

```rust
escaped(take_till0("([{},|\\"), '\\', one_of("()[]{},|\\"))
```

### Results

- **expected**: The whole text is matched by this parser
- **actual**: It is only matched up to `\`

```
> TRACE_ROOT "foo \| bar"   | "foo \\| bar"∅
 > context="balanced_text"  | "foo \\| bar"∅
  > escaped                 | "foo \\| bar"∅
   > take_till0             | "foo \\| bar"∅
   < take_till0             | +4
   > take_till0:1           | "\\| bar"∅
   < take_till0:1           | +0
  < escaped                 | +4
 < context="balanced_text"  | +4
< TRACE_ROOT "foo \| bar"   | +4
> eof                       | "\\| bar"∅
< eof   
```

### Question

What am I doing wrong here?

**Comment by epage** [comment, marked as answer]
I believe you need `take_till1` instead of `take_till0`.

We only fallback to escaping if the regular parser fails.  Instead, we see the regular parser returning nothing and assume the input is done.  I can see calling this out more explicitly in the docs

**Reply by nicopap** [reply]
Using `take_till1` did indeed solve this issue! Thank you very much!

Thank you for making winnow. I had difficulties with `nom`'s error type and understanding what the parsers were doing, `winnow` was exactly what I needed.
